cmd/sovereignnode/chainSimulator/sovereignChainSimulator.go

```diff
@@ -63,9 +67,11 @@ func NewSovereignChainSimulator(args ArgsSovereignChainSimulator) (chainSimulato
 		return rating.NewSovereignRatingsData(arg)
 	}
 	args.CreateIncomingHeaderSubscriber = func(config *config.NotifierConfig, dataPool dataRetriever.PoolsHolder, mainChainNotarizationStartRound uint64, runTypeComponents factory.RunTypeComponentsHolder) (process.IncomingHeaderSubscriber, error) {
+		config = &alteredSovConfigs.NotifierConfig
```

Comment:
This is not needed because the config comes from func args and is already altered.

Reply:
+1

Reply:
That's not the case here. These functions are called before `AlterConfig` is called further down in base chain simulator, so I need the altered configs beforehand. Test would not pass anyway otherwise.

Reply:
No, this `CreateIncomingHeaderSubscriber` is called after `AlterConfig`.
AlterConfig is the first thing, then is Create Core comps, Crypto comps, ..., IncomingHeader, and so on

Reply:
Not the case for runType comps, those are called before, please check for yourself, I need that extra variable and to change it. Maybe not also for incoming, but definetely for runType